addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -6,61 +6,76 @@
+    l10n_it_edi_checkbox_send = fields.Boolean('Tax Agency (Italy)',
+        compute='_compute_l10n_it_edi_checkbox_send',
+        store=True,
+        readonly=False,
+        help="Send the e-invoice XML to the Italian Tax Agency.")
 
     # -------------------------------------------------------------------------
     # COMPUTE/CONSTRAINS METHODS
     # -------------------------------------------------------------------------
 
-    @api.depends('l10n_it_edi_enable_send')
-    def _compute_send_mail_extra_fields(self):
-        # EXTENDS account
-        super()._compute_send_mail_extra_fields()
+    @api.depends('move_ids')
+    def _compute_l10n_it_edi_xml_export(self):
         for wizard in self:
-            wizard.l10n_it_edi_enable_send = any(
-                wizard._get_default_l10n_it_edi_enable_send(m)
-                and not m.l10n_it_edi_attachment_id
-                for m in wizard.move_ids)
-
+            already_has_xml = any(wizard.move_ids.mapped("l10n_it_edi_attachment_id"))
+            already_has_pdf = any(wizard.move_ids.mapped("invoice_pdf_report_id"))
             if not wizard.company_id.l10n_it_edi_proxy_user_id:
                 wizard.l10n_it_edi_warning_message = _("You must accept the terms and conditions in the Settings to use the IT EDI.")
             else:
                 wizard.l10n_it_edi_warning_message = wizard.move_ids._l10n_it_edi_format_export_data_errors()
+            wizard.l10n_it_edi_enable_xml_export = any(m._l10n_it_edi_ready_for_xml_export() for m in wizard.move_ids)
+            wizard.l10n_it_edi_readonly_xml_export = bool(wizard.l10n_it_edi_warning_message) or already_has_pdf or already_has_xml
 
-            already_has_pdf = any(wizard.move_ids.mapped("invoice_pdf_report_id"))
-            already_has_xml = any(x._is_l10n_it_edi_import_file() for x in wizard.move_ids.mapped("attachment_ids"))
-            wizard.l10n_it_edi_readonly = wizard.l10n_it_edi_warning_message or already_has_pdf or already_has_xml
+    @api.depends('move_ids', 'l10n_it_edi_checkbox_xml_export', 'l10n_it_edi_warning_message')
+    def _compute_l10n_it_edi_enable_readonly_send(self):
+        for wizard in self:
+            already_has_xml = any(wizard.move_ids.mapped("l10n_it_edi_attachment_id"))
+            xml_already_sent = all(m.l10n_it_edi_state not in (False, 'rejected') for m in wizard.move_ids)
+            wizard.l10n_it_edi_enable_send = already_has_xml or wizard.l10n_it_edi_checkbox_xml_export
+            wizard.l10n_it_edi_readonly_send = bool(wizard.l10n_it_edi_warning_message) or xml_already_sent
 
-    @api.depends('l10n_it_edi_readonly', 'l10n_it_edi_enable_send')
+    @api.depends('move_ids')
+    def _compute_l10n_it_edi_checkbox_xml_export(self):
+        for wizard in self:
+            wizard.l10n_it_edi_checkbox_xml_export = wizard.l10n_it_edi_enable_xml_export and not wizard.l10n_it_edi_readonly_xml_export
```

Comment:
It works by chance that this method gets calculated after?  (we could just recalculate the values of those fields)

Reply:
It's already computed.

Reply:
Indeed, the compute could be triggered as it is only calculated at the start so it should be fine